src/pages/LogDrop/state.ts

```diff
@@ -0,0 +1,58 @@
+import { useReducer } from "react";
+
+type CurrentState =
+  | "WAITING_FOR_FILE"
+  | "LOADING_FILE"
+  | "PROMPT_FOR_PARSING_METHOD"
+  | "FILE_ERROR";
+
+type State = {
+  currentState: CurrentState;
+  file: File | null;
+  parsingMethod: string | null;
+  error: Error | null;
+  fileName: string | null;
+};
+type Action =
+  | { type: "DROPPED_FILE"; file: File }
+  | { type: "PARSE_FILE" }
+  | { type: "CANCEL" };
+
+const initialState = (): State => ({
+  currentState: "WAITING_FOR_FILE",
+  file: null,
+  parsingMethod: null,
+  error: null,
+  fileName: null,
+});
+
+const reducer = (state: State, action: Action): State => {
+  switch (action.type) {
+    case "DROPPED_FILE":
+      return {
+        ...state,
+        currentState: "PROMPT_FOR_PARSING_METHOD",
+        file: action.file,
+        fileName: action.file.name,
```

Comment:
what's the reason to store the `fileName` separately if it can be accessed via the file?

Reply:
Thanks this was left in by accident.